### ISSUE TITLE
#NumberRange only visible in debugging/logging?

### ISSUE THREAD

**Issue by Jakeii** [spencermountain/compromise#684]
I have seen `#NumberRange` in the verbose logging output, and throughout the source code, but can't seem to be able to use to it in any way. Possibly a bug?

I noticed it also gets tagged as a `#Value` in the verbose log but then disappears

I suspect it's because the contraction `1-2` gets expanded to `1 to 2` so the original tags get lost somewhere?

```
// verbose
> nlp('1-2 kg')
 kg   + Unit 
 kg   + → Noun 
1-2  + Value 
1-2  + NumberRange 
1-2  + → Contraction 
 kg   + Singular 
 kg   + → Noun 
 to   + Conjunction 
 
> nlp('1-2 kg').debug()
'[1]'     -  Number, Cardinal
'[to]'    -  Conjunction
'[2]'     -  Number, Cardinal
'kg'      -  Unit, Noun, Singular

> nlp('1-2 kg').has('#NumberRange')
false

> nlp('1-2 kg').match('#NumberRange').text()
""
```

**Comment by Drache93** [comment]
Good find! Yeh looks like a bug in the contraction handling. `1 to 2` works fine when used directly.

**Comment by Jakeii** [comment]
Is it as simple as checking in the contraction handler if it is a range and adding the phrase rather than replacing? I could open a PR in that case because it appears to work.

**Comment by Drache93** [comment]
Sorry not had a chance to look today. Yeh that should be it, @spencermountain will know more, there's probably a good reason why the tags are replaced rather than amended. So this is probably the exception rather than the rule. If you make a PR, v13 I think will be this week.

**Comment by spencermountain** [comment]
great find. Thanks Jake. 
Yeah, the logic about keeping existing tags during .replace() changed in v12. 
Good observation that it got tagged as a NumberRange briefly.
got it fixed on dev. And as Dominic said, it should be released this week.
cheers

**Comment by Jakeii** [comment]
Thanks guys!

**Comment by spencermountain** [comment]
fixed in `v13.0.0`
